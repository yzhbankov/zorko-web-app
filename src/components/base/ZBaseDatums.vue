<template>
  <v-flex xs12>
    <v-list two-line subheader>
      <ZBaseDatumsItem
        v-for="item in items"
        :key="item.name"
        :name="item.name"
        :title="item.title"
        :connection="item.connection"
        :description="item.description"
        :info="item.info"
        :looksCount="item.looksCount"
        @datumActivated="onDatumActivated"
      />
    </v-list>
  </v-flex>
</template>

<script>
import ZBaseDatumsItem from './ZBaseDatumsItem.vue'

export default {
  name: 'ZBaseDatums',
  components: {
    ZBaseDatumsItem
  },
  props: {
    items: {
      type: Array
    }
  },
  methods: {
    onDatumActivated (item) {
      this.$emit('datumInput', item)
    }
  }
}
</script>
