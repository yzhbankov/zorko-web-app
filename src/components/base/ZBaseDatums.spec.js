import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import ZBaseDatums from './ZBaseDatums.vue'
import ZBaseDatumsItem from './ZBaseDatumsItem.vue'

import mockDatums from '../../../static/__mocks__/datums.json'

describe('ZBaseDatums.vue', () => {
  let store
  let propsData

  beforeEach(() => {
    propsData = {
      items: mockDatums
    }
  })

  it('has proper amount of items', () => {
    const wrapper = shallow(ZBaseDatums, {
      store,
      propsData
    })
    const items = wrapper.findAll(ZBaseDatumsItem)

    expect(items.length).toEqual(2)
  })

  it('emits `datumInput`', () => {
    const wrapper = shallow(ZBaseDatums, {
      store,
      propsData
    })
    const {vm} = wrapper.find(ZBaseDatumsItem)

    vm.$emit('datumActivated', 'baz')

    expect(wrapper.emitted('datumInput')[0][0]).toEqual('baz')
  })

  it('has same HTML for looks', () => {
    const renderer = createRenderer()
    const homeWrapper = shallow(ZBaseDatums, {
      store,
      propsData
    })

    renderer.renderToString(homeWrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
