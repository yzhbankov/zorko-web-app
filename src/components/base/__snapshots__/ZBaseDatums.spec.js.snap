// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZBaseDatums.vue has same HTML for looks 1`] = `
<v-flex xs12="">
    <v-list two-line="" subheader="">
        <!---->
        <!---->
    </v-list>
</v-flex>
`;
